lib/jsonapi_parameters/stack_limit.rb

```diff
@@ -0,0 +1,44 @@
+module JsonApi
+  module Parameters
+    LIMIT = 3
+
+    class StackLevelTooDeep < StandardError
+    end
+
+    def stack_limit=(val)
+      @stack_limit = val
+    end
+
+    def stack_limit
+      @stack_limit ||= LIMIT
```

Comment:
>about we move it to the class initializer

There is no class initializer though, as JsonApi::Parameters is a module overall. I tried to use some `self.included(base)` do/end block to define the initializer but sadly with no success. It has to work both for core extension (`ActionController::Parameters`) and just for a basic Ruby class when the module is included. I think I forgot how to do that part of Ruby, I gotta dig a bit deeper and see how to make that work.

Reply:
Ok, you are right. It's no so easy.
Perhaps we could add an explicit initialiation method called in `lib/jsonapi_parameters/translator.rb` on line 7
Instead of 
```
self.stack_limit = custom_stack_limit
```
there could be smth like 
```
initialize_stack(custom_stack_limit) # this would also include `stack_level` initialization
```

I'm not sure are there other places where it would be needed, but I think there not that many entry points to the jsonapi processing, and those would have to be updated.

Having said that, I won't be that picky about it (as you have a much better knowledge of the gem). So I would leave it to you at this point. If you think the above idea is not valid for whatever reason, we can just go ahead and approve this PR as I have no other comments at this point.